pkg/controller/keystone/keystone_controller.go

```diff
@@ -120,12 +120,26 @@ func (r *ReconcileKeystone) Reconcile(request reconcile.Request) (reconcile.Resu
 		return reconcile.Result{}, nil
 	}
 
+	keystoneService := newKeystoneService(keystone)
```

Comment:
this and createOrUpdate could be in method createKeystoneService Reconcile method would look much much better and more readable, but that's a nit